modules/openapi-generator/src/main/resources/python/api_doc_example.mustache

```diff
@@ -5,24 +5,17 @@ import {{{packageName}}}
 from {{{packageName}}}.rest import ApiException
 from pprint import pprint
 {{> python_doc_auth_partial}}
-{{#hasAuthMethods}}
-# Defining host is optional and default to {{{basePath}}}
-configuration.host = "{{{basePath}}}"
 # Enter a context with an instance of the API client
+{{#hasAuthMethods}}
```

Comment:
Some endpoints need to instantiate a configuration instance, set auth method info on it, and then use that configuration when instantiating the api_client.
For each endpoint, can you loop through authmethods, select the first one, set the relevant configuration for it, and then use it?
```
{{#hasAuthMethods}}
{{authMethods}}
{{#-first}}
{{isBasicBasic}}
config = blah
{{/isBasicBasic}}
{{isBasicBearer}}
config = blah
{{/isBasicBearer}}
{{isHttpSignature}}
config = blah
{{/isHttpSignature}}
{{/-first}}
{{/authMethods}}
with {{{packageName}}}.ApiClient(configuration) as api_client:
{{/hasAuthMethods}}
{{^hasAuthMethods}}
with {{{packageName}}}.ApiClient() as api_client:
{{/hasAuthMethods}}
```

Reply:
I think most of that is handled already by the [`python_doc_auth_partial`](https://github.com/OpenAPITools/openapi-generator/pull/5161/files#diff-58886cfb800110cb8aa3236113fb91e0R7) fragment, content of that fragment is here: https://github.com/OpenAPITools/openapi-generator/blob/20f9aee16c1b0e59ceb0b4b8fada5be7c0deeaf3/modules/openapi-generator/src/main/resources/python/python_doc_auth_partial.mustache

Reply:
Ah, I missed that, sorry. Thank you for pointing me there. This is good as-is.